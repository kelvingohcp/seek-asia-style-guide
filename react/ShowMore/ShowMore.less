* {
  box-sizing: border-box;
}

.root {
  position: relative;
  margin: 2em auto;
  width: 70%;
}

.panel {
  overflow: hidden;
  transition: max-height 0.5s ease;
}
.fade {
  height: 100px;
  margin-top: -100px;
  position: relative;
}
.button{
  width: 100%;
}
