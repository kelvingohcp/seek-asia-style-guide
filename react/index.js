export { default as StyleGuideProvider } from './StyleGuideProvider/StyleGuideProvider';
export { default as SeekApp } from './StyleGuideProvider/StyleGuideProvider'; // Alias for backwards compatibility

// Components
export { default as Alert } from './Alert/Alert';
export { default as Button } from './Button/Button';
export { default as FilterButton } from './FilterButton/FilterButton';
export { default as ButtonGroup } from './ButtonGroup/ButtonGroup';
export { default as ListItem } from './ListItem/ListItem';
export { default as Loader } from './Loader/Loader';
export { default as Logo } from './Logo/Logo';
export { default as LogoRainbow } from './LogoRainbow/LogoRainbow';
export { default as Rating } from './Rating/Rating';
export { default as NavigationBar } from './NavigationBar/NavigationBar';
export { default as JobCard, trackLinkType } from './JobCard/JobCard';
export { default as JobCardLoader } from './JobCardLoader/JobCardLoader';
export { default as CardLoader } from './CardLoader/CardLoader';
export { default as CardLoadingBar } from './CardLoadingBar/CardLoadingBar';
export { default as Header } from './Header/Header';
export { default as ShowMore } from './ShowMore/ShowMore';
export { default as TitledCard } from './TitledCard/TitledCard';
export { default as Icon } from './Icon/Icon';
export { default as DropdownList } from './DropdownList/DropdownList';
export { default as Tag } from './Tag/Tag';
export { default as Badge } from './Badge/Badge';

// Typography
export { default as Critical } from './Critical/Critical';
export { default as Paragraph } from './Paragraph/Paragraph';
export { default as Positive } from './Positive/Positive';
export { default as Secondary } from './Secondary/Secondary';
export { default as Strong } from './Strong/Strong';
export { default as Text } from './Text/Text';
export { default as TextLink } from './TextLink/TextLink';

// Layout
export { default as AsidedLayout } from './AsidedLayout/AsidedLayout';
export { default as Card } from './Card/Card';
export { default as CardGroup } from './CardGroup/CardGroup';
export { default as Columns } from './Columns/Columns';
export { default as Hidden } from './Hidden/Hidden';
export { default as PageBlock } from './PageBlock/PageBlock';
export { default as Section } from './Section/Section';

// Icons
export { default as CriticalIcon } from './CriticalIcon/CriticalIcon';
export { default as CrossIcon } from './CrossIcon/CrossIcon';
export { default as ErrorIcon } from './ErrorIcon/ErrorIcon';
export { default as FieldOfStudyIcon } from './FieldOfStudyIcon/FieldOfStudyIcon';
export { default as FlagHKIcon } from './FlagHKIcon/FlagHKIcon';
export { default as FlagIDIcon } from './FlagIDIcon/FlagIDIcon';
export { default as FlagMYIcon } from './FlagMYIcon/FlagMYIcon';
export { default as FlagPHIcon } from './FlagPHIcon/FlagPHIcon';
export { default as FlagSGIcon } from './FlagSGIcon/FlagSGIcon';
export { default as FlagTHIcon } from './FlagTHIcon/FlagTHIcon';
export { default as FlagVNIcon } from './FlagVNIcon/FlagVNIcon';
export { default as HelpIcon } from './HelpIcon/HelpIcon';
export { default as InfoIcon } from './InfoIcon/InfoIcon';
export { default as JobFunctionIcon } from './JobFunctionIcon/JobFunctionIcon';
export { default as JobInvitationIcon } from './JobInvitationIcon/JobInvitationIcon';
export { default as MoreIcon } from './MoreIcon/MoreIcon';
export { default as ProfileIcon } from './ProfileIcon/ProfileIcon';
export { default as StarIcon } from './StarIcon/StarIcon';
export { default as TickCircleIcon } from './TickCircleIcon/TickCircleIcon';

// Fields
export { default as Autosuggest } from './Autosuggest/Autosuggest';
export { default as Checkbox } from './Checkbox/Checkbox';
export { default as ClearField } from './ClearField/ClearField';
export { default as Dropdown } from './Dropdown/Dropdown';
export { default as EmailField } from './EmailField/EmailField';
export { default as MonthPicker } from './MonthPicker/MonthPicker';
export { default as SlideToggle } from './SlideToggle/SlideToggle';
export { default as Textarea } from './Textarea/Textarea';
export { default as TextField } from './TextField/TextField';

// Accessibility
export { default as ScreenReaderOnly } from './ScreenReaderOnly/ScreenReaderOnly';
export { default as ScreenReaderSkipLink } from './ScreenReaderSkipLink/ScreenReaderSkipLink';
export { default as ScreenReaderSkipTarget } from './ScreenReaderSkipTarget/ScreenReaderSkipTarget';

// Constants
export { default as Constants } from './Constants/Constants';
