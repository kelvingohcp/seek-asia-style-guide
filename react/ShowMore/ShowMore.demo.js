import React from 'react';
import ShowMore from './ShowMore';
import { Card, Section, Text } from 'seek-asia-style-guide/react';
import * as sketch from './ShowMore.sketch';

export default {
  route: '/showMore',
  title: 'Show More',
  component: ShowMore,
  sketch,
  initialProps: {
    children: (
      <Card>
        <Section>
          <Text shouting>
            At jobsDB, we believe in an open and inclusive world for all.
          </Text>
          <Text>
            Here, opportunities and possibilities are plenty: A fresh start. A
            new career.<br />
            A welcome change. These are just a few of the things we help people
            explore.<br />
            As a leading job portal with substantial positions across China,
            Hong Kong, Indonesia, Philippines, Singapore and Thailand, we are
            Asia’s preferred destination for job seekers.<br />With a team of
            1,000 talented individuals, we’ve a strong local presence in every
            country we’re in.<br />
            With our understanding and insights on each market, as well our
            passion for technology, we provide job seekers the best
            opportunities and deliver exceptional value to employers.<br /> By
            really understanding our job seekers and employers, we ensure they
            find what’s right for them.<br /> After all, we’re not just about
            filling vacancies but fulfilling true potential.<br />
            e became part of the Australian Stock Exchange-listed SEEK Limited –
            the world’s largest job portal by market capitalisation.<br /> As
            part of the SEEK family, we leverage world-class products to match
            talented job seekers with reputable employers across the region. Of
            course, there’s always more to do.<br /> That’s why we continue to
            explore new frontiers across all online channels.<br /> So we can
            have the right person for the right job. There’s better out there.
            Here’s where you find it.<br />
          </Text>
        </Section>
      </Card>
    ),
    showLessHeight: 100,
    lblShowMore: 'Show more',
    lblShowLess: 'Show less'
  },
  options: [
    {
      label: 'Component',
      type: 'radio',
      states: [
        {
          label: '150px height',
          transformProps: props => ({
            ...props,
            showLessHeight: 150
          })
        },
        {
          label: '300px Height',
          transformProps: props => ({
            ...props,
            showLessHeight: 300
          })
        },
        {
          label: '700px Height',
          transformProps: props => ({
            ...props,
            showLessHeight: 700
          })
        }
      ]
    },
    {
      label: 'disable',
      type: 'checklist',
      states: [
        {
          label: 'disable',
          transformProps: props => ({
            ...props,
            disable: true
          })
        }
      ]
    },
    {
      label: 'onPanelToggle',
      type: 'checklist',
      states: [
        {
          label: 'onPanelToggle',
          transformProps: props => ({
            ...props,
            onPanelToggle: ({ status }) => {
              alert(status); // eslint-disable-line no-alert
            }
          })
        }
      ]
    },
    {
      label: 'onPanelOpen',
      type: 'checklist',
      states: [
        {
          label: 'onPanelOpen',
          transformProps: props => ({
            ...props,
            onPanelOpen: () => {
              alert('hi'); // eslint-disable-line no-alert
            }
          })
        }
      ]
    }
  ]
};
