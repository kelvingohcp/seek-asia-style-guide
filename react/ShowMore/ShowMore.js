import styles from './ShowMore.less';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, ChevronIcon } from 'seek-asia-style-guide/react';
import classnames from 'classnames';

export default class ShowMore extends Component {
  static propTypes = {
    component: PropTypes.function,
    showLessHeight: PropTypes.number,
    lblShowMore: PropTypes.string,
    lblShowLess: PropTypes.string,
    disable: PropTypes.boolean
  };

  static defaultProps = {
    showLessHeight: 50,
    lblShowMore: 'Show more',
    lblShowLess: 'Show less'
  };

  constructor(props) {
    super(props);

    this.state = {
      contentHeight: 0,
      isPanelOpened: false
    };
    this.handleClick = this.handleClick.bind(this);
    this.updateDimensions = this.updateDimensions.bind(this);
    this.myRef = React.createRef();
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateDimensions);
    this.updateDimensions();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateDimensions);
  }

  updateDimensions() {
    this.setState({
      contentHeight: this.myRef.current.clientHeight
    });
  }

  handleClick(e) {
    e.preventDefault();
    this.setState({
      isPanelOpened: !this.state.isPanelOpened
    });
  }

  render() {
    const panelHeight = (this.state.isPanelOpened || this.props.disable) ? this.state.contentHeight : this.props.showLessHeight;
    return (
      <div className={classnames({ [styles.root]: true })}>
        <div
          className={classnames(styles.panel)}
          style={{ maxHeight: `${panelHeight}px` }} >
          <div ref={this.myRef}>{this.props.component}</div>
        </div>
        {
          this.props.disable ? false : this.state.contentHeight > this.props.showLessHeight && (
            <Button
              id='btnShowMore'
              color='hyperlink'
              className={classnames(styles.button)}
              onClick={this.handleClick} >
              <span>
                {this.state.isPanelOpened ? this.props.lblShowLess : this.props.lblShowMore}
              </span>
              <span> <span><ChevronIcon direction={this.state.isPanelOpened ? 'up' : 'down'} /></span></span>
            </Button>
          )
        }
      </div >
    );
  }
}
