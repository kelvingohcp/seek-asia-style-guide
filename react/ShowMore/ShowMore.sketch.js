import React from 'react';
import mapKeys from 'lodash/mapKeys';
import ShowMore from '../ShowMore/ShowMore';
import { Card, Section, Text } from 'seek-asia-style-guide/react';

const card = <Card><Section><Text>At jobsDB, we believe in an open and inclusive world for all.Here, opportunities and possibilities are plenty: A fresh start. A new career.A welcome change. These are just a few of the things we help people explore.</Text></Section></Card>;
export const showMore = {
  '150px': <ShowMore showLessHeight='150'>{card}</ShowMore>,
  '50px': <ShowMore showLessHeight='50'>{card}</ShowMore>
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(showMore, (value, key) => `ShowMore/${key}`);
