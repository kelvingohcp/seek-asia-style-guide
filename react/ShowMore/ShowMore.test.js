import React from 'react';
import renderer from 'react-test-renderer';
import ShowMore from './ShowMore';
import { Card, Section, Text } from 'seek-asia-style-guide/react';

const card = (
  <Card>
    <Section>
      <Text>
        At jobsDB, we believe in an open and inclusive world for all.Here,
        opportunities and possibilities are plenty: A fresh start. A new
        career.A welcome change. These are just a few of the things we help
        people explore.
      </Text>
    </Section>
  </Card>
);

function createNodeMock() {
  return card;
}

describe('ShowMore:', () => {
  it('should render value', () => {
    const options = { createNodeMock };
    const wrapper = renderer.create(
      <ShowMore
        showLessHeight='150'
        lblShowMore='Show more'
        lblShowLess='Show less'>
        {card}
      </ShowMore>,
      options
    );
    expect(wrapper).toMatchSnapshot();
  });
});
