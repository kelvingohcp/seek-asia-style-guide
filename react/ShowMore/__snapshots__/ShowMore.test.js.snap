// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMore: should render value 1`] = `
<div
  className="root"
>
  <div
    className="panel"
    style={
      Object {
        "maxHeight": "150px",
      }
    }
  >
    <div />
  </div>
</div>
`;
